ui/src/features/Editor/components/OverlayUI/components/DebugPanel/index.tsx

```diff
@@ -0,0 +1,218 @@
+import {
+  CaretDownIcon,
+  CaretUpIcon,
+  CodeIcon,
+  CornersInIcon,
+  CornersOutIcon,
+  EyeIcon,
+  MinusIcon,
+  TerminalIcon,
+} from "@phosphor-icons/react";
+import { memo, useEffect, useRef, useState } from "react";
+
+import {
+  ResizableHandle,
+  ResizablePanel,
+  ResizablePanelGroup,
+  Select,
+  SelectContent,
+  SelectItem,
+  SelectTrigger,
+  SelectValue,
+  Tabs,
+  TabsContent,
+  TabsList,
+  TabsTrigger,
+} from "@flow/components";
+import { useT } from "@flow/lib/i18n";
+
+import DebugLogs from "./DebugLogs";
+import DebugPreview from "./DebugPreview";
+import { DataTable } from "./DebugPreview/components";
+import useHooks from "./hooks";
+
+type Props = {
+  fullscreenDebug: boolean;
+  onFullScreenExpand: () => void;
+};
+
+const DebugPanel: React.FC<Props> = ({
+  fullscreenDebug,
+  onFullScreenExpand,
+}) => {
+  const {
+    debugJobId,
+    debugJobState,
+    fileType,
+    expanded,
+    minimized,
+    showTempPossibleIssuesDialog,
+    selectedDataURL,
+    dataURLs,
+    selectedOutputData,
+    isLoadingData,
+    handleExpand,
+    handleMinimize,
+    handleTabChange,
+    handleShowTempPossibleIssuesDialogClose,
+    handleSelectedDataChange,
+  } = useHooks();
+  const t = useT();
+  const [tabValue, setTabValue] = useState("debug-logs");
+  const hasSwitchedToViewerRef = useRef(false);
+  const debugJobIdRef = useRef(debugJobId);
+
+  useEffect(() => {
+    if (debugJobId !== debugJobIdRef.current) {
+      debugJobIdRef.current = debugJobId;
+    }
+
+    if (dataURLs && !hasSwitchedToViewerRef.current) {
+      setTabValue("debug-viewer");
+      hasSwitchedToViewerRef.current = true;
+    }
+
+    if (!dataURLs) {
+      setTabValue("debug-logs");
+      hasSwitchedToViewerRef.current = false;
+    }
+  }, [dataURLs, debugJobId]);
+
+  return debugJobId ? (
+    <Tabs
+      className={`pointer-events-auto w-[95vw] overflow-hidden rounded-md bg-secondary shadow-md shadow-secondary transition-all ${minimized ? "h-[36px]" : fullscreenDebug ? "h-[100vh] w-[100vw]" : expanded ? "h-[80vh]" : "h-[500px]"}`}
+      value={tabValue}
+      defaultValue="debug-logs"
+      onDoubleClick={handleExpand}
+      onValueChange={setTabValue}>
+      <div className={`flex items-center ${minimized ? "" : "border-b"} p-1`}>
+        <div className="relative flex w-fit items-center p-1">
```

Comment:
Here again a random relative with flex on a div that wraps only a single other div. Doesn't make sense